core/src/test/java/org/elasticsearch/common/util/concurrent/ThreadContextTests.java

```diff
@@ -109,6 +110,57 @@ public void testStoreContext() {
         assertNull(threadContext.getHeader("foo.bar"));
     }
 
+    public void testRestorableContext() {
+        Settings build = Settings.builder().put("request.headers.default", "1").build();
+        ThreadContext threadContext = new ThreadContext(build);
+        threadContext.putHeader("foo", "bar");
+        threadContext.putTransient("ctx.foo", 1);
+        Supplier<ThreadContext.StoredContext> contextSupplier = threadContext.newRestorableContext(true);
```

Comment:
shall we add a duplicate response header too? I.e., one that is both in the original context and the one being replcaed?